modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -385,15 +385,25 @@ export class {{classname}} {
 
 {{/hasFormParams}}
 {{#useHttpClient}}
+    {{^isResponseFile}}
+        let responseType: any = undefined;
```

Comment:
I am concerned that we lose type-checking for the httpClient method invocation, since it uses the exact value of `responseType` (`json`, `text`, `arraybuffer`, `blob`) to determine the correct method signature, see e.g.
https://github.com/angular/angular/blob/168abc6d6f52713383411b14980e104c99bfeef5/packages/common/http/src/client.ts#L963-L970

a better way to achieve this would be to implement this check in https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java and export a variable (similar to `isResponseFile`) that can be checked in the template to generate either
```
responseType: 'text'
```
or
```
responseType: 'blob'
```
or nothing (for the json case)

@wing328 maybe there is already a variable that tells whether we expect a text response?

Reply:
The problem with this is the 'Accept' header is determined at run time. The configuration object supports a user-defined function for choosing an 'Accept' header making the response type unpredictable (at least for operations supporting multiple content types). This could be mitigated by explicitly providing the desired content type in the function parameters and generate typescript definitions for each content type.